src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs

```diff
@@ -33,6 +37,22 @@ public AuthorizationMiddleware(RequestDelegate next, IAuthorizationPolicyProvide
     {
         _next = next ?? throw new ArgumentNullException(nameof(next));
         _policyProvider = policyProvider ?? throw new ArgumentNullException(nameof(policyProvider));
+        _canCache = false;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of <see cref="AuthorizationMiddleware"/>.
+    /// </summary>
+    /// <param name="next">The next middleware in the application middleware pipeline.</param>
+    /// <param name="policyProvider">The <see cref="IAuthorizationPolicyProvider"/>.</param>
+    /// <param name="services">The <see cref="IServiceProvider"/>.</param>
+    public AuthorizationMiddleware(RequestDelegate next, IAuthorizationPolicyProvider policyProvider, IServiceProvider services) : this(next, policyProvider)
+    {
+        if (services != null && _policyProvider.CanCachePolicy)
```

Comment:
services can't be null.

Reply:
Add ArgumentNullException.ThrowIfNull for it